From: Egbert Eich <eich@suse.com>
Date: Mon May 13 16:52:11 2019 +0200
Subject: Handle zypper error code correctly
Patch-mainline: Not yet
Git-commit: 0fbc15393d0f615ef6ad2e70837430817ab72992
References: 

Do not consider installation scriptlet failures as an installation
failure.

Signed-off-by: Egbert Eich <eich@suse.com>
---
 .../singularity/internal/pkg/build/sources/conveyorPacker_zypper.go | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)
diff --git a/src/github.com/sylabs/singularity/internal/pkg/build/sources/conveyorPacker_zypper.go b/src/github.com/sylabs/singularity/internal/pkg/build/sources/conveyorPacker_zypper.go
index 0e6f00a..d08d61d 100644
--- a/src/github.com/sylabs/singularity/internal/pkg/build/sources/conveyorPacker_zypper.go
+++ b/src/github.com/sylabs/singularity/internal/pkg/build/sources/conveyorPacker_zypper.go
@@ -113,7 +113,11 @@ func (cp *ZypperConveyorPacker) Get(b *types.Bundle) (err error) {
 
 	// run zypper
 	if err = cmd.Run(); err != nil {
-		return fmt.Errorf("While bootstrapping from zypper: %v", err)
+		if ret, _ := system.GetExitCode(err); ret == 107 {
+		       sylog.Warningf("Bootstrap succeeded, some RPM scripts failed");
+		} else {
+		       return fmt.Errorf("While bootstrapping from zypper: %v", err)
+		}
 	}
 
 	return nil
